modules/flowable-engine/src/test/java/org/flowable/engine/test/api/runtime/RuntimeActivityInstanceTest.java

```diff
@@ -420,44 +415,47 @@ public void testLoop() {
             org.flowable.task.api.Task task = taskService.createTaskQuery().processInstanceId(processInstance.getId()).singleResult();
             Number inputNumber = (Number) taskService.getVariable(task.getId(), "input");
             int input = inputNumber.intValue();
-            assertEquals(i, input);
+            assertThat(input).isEqualTo(i);
             taskService.complete(task.getId(), CollectionUtil.singletonMap("input", input + 1));
             task = taskService.createTaskQuery().processInstanceId(processInstance.getId()).singleResult();
         }
-        
+
         waitForHistoryJobExecutorToProcessAllJobs(7000, 100);
 
         // Verify history
         List<ActivityInstance> taskActivityInstances = runtimeService.createActivityInstanceQuery().activityType("userTask").list();
-        assertEquals(10, taskActivityInstances.size());
-        for (ActivityInstance activityInstance : taskActivityInstances) {
-            assertNotNull(activityInstance.getStartTime());
-            assertNotNull(activityInstance.getEndTime());
-        }
+        assertThat(taskActivityInstances).hasSize(10);
+        assertThat(taskActivityInstances)
+                .extracting(ActivityInstance::getEndTime, ActivityInstance::getEndTime)
+                .doesNotContainNull();
```

Comment:
Using `doesNotContainNull` here is incorrect. It will always be not `null` since the assertion is on the object which is a `tuple` of not null values. You can use `flatExtracting` or keep the loop.

Same comment applies to the other usages of `doesNotContainNull` that extract more than one value